### ISSUE TITLE
run('echo something') on Windows shows exit 0 appended

### ISSUE THREAD

**Issue by ck81** [tebelorg/TagUI-Python#45]
Hi Ken,

In windows 10, the following works:
```
run_result = t.run("echo %SYSTEMROOT%")
print(run_result)
```

Result is: `C:\WINDOWS; exit 0`

While it's not difficult in python to use some function to remove that `"; exit 0"` at the end, thought it might be more convenient for users if TagUI-Python does not return that in the result string.

**Comment by kensoh** [comment]
Hi CK, thanks for raising this! May need your help to test because I don't have access to a Windows laptop now. Issues #45 #46 should be related to the same cause.

The best practice for running commands from Python using subprocess requires using the exit 0 to be always appended. This is to capture error messages if the command results in error -
https://docs.python.org/3.4/library/subprocess.html?highlight=exp#subprocess.check_output

Below shows a replication case if exit 0 is not appended and what happens if appended. When it is not appended as part of the command, if the command returns error it will raise a Python exception with error code instead of showing the error message which is more meaningful for debugging.

```
>>> t.run('abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "tagui.py", line 1173, in run
    shell=True))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'abc' returned non-zero exit status 127

>>> t.run('abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "tagui.py", line 1173, in run
    shell=True))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command 'abc' returned non-zero exit status 127
>>> t.run('abc; exit 0')
'/bin/sh: abc: command not found\n'
```

Will dig further. There may not be a solution or the solution could be not appending for echo commands. For the cmd error, looks like it is some other cause.

**Comment by kensoh** [comment]
For echo on macOS, it looks alright with appending exit 0. 

```
>>> t.run('echo 123; exit 0')
'123\n'
```

**Comment by kensoh** [comment]
On macOS, the ; works in echo as a delimiter for next command. From your above example, it looks like for Windows, the ; is not recognised by echo as a delimiter. Thus the `;exit 0` string becomes part of the echo output result when you run echo.

```
Tebel:tagui-python kensoh$ echo 123; echo 456
123
456
```

**Comment by kensoh** [comment]
Solution may be to use `&` when run() is invoked on Windows environment. I'll have to borrow a Windows laptop and test on this.

https://stackoverflow.com/questions/8055371/how-do-i-run-two-commands-in-one-line-in-windows-cmd

**Comment by ck81** [comment]
Hi Ken,

> On macOS, the ; works in echo as a delimiter for next command. 

Yes, you're right. The above works for Mac but not on windows.

On windows 10, `t.run('echo 123; exit 0')` gives
```
'123; exit 0; exit 0\r\n'
```

`t.run('echo 123 & exit 0')` gives
```
'123 \r\n'
```

Think you can close this issue.  It's not something really important, and it's easy in Python to use some function to remove that `"; exit 0"`  But you might want to mention this somewhere in your documentation so that people are aware of this when they port applications between TagUI and TagUI-Python.

**Comment by kensoh** [comment]
Thanks CK for your update and confirmation. Oh this should be fixed where possible, I think it is the same root cause of the issue as #46 - probably the ; gets interpreted as part of the command resulting in the error message, will check on this when I borrow a Windows laptop.

**Comment by kensoh** [comment]
Going to update run() to below, tested ok on Windows. Will test on macOS and Linux before patching -

```py
def run(command_to_run = None):
    if command_to_run is None or command_to_run == '':
        print('[TAGUI][ERROR] - command(s) missing for run()')
        return ''

    else:
        if platform.system() == 'Windows':
            command_delimiter = ' & '
        else:
            command_delimiter = '; '
        return _py23_decode(subprocess.check_output(
            command_to_run + command_delimiter + 'exit 0',
            stderr=subprocess.STDOUT,
            shell=True))
```

**Comment by kensoh** [comment]
Closing #45 as it is a duplicate issue with same root cause as #46. See #46 for updated details.

**Comment by kensoh** [comment]
Fix implemented in v1.10, details in #46. `pip install tagui --upgrade` for latest version.
